### PR TITLE
[configuration-validation]: convert to typescript (release 1.0.0)

### PR DESCRIPTION
**Pull request by aarongranick-okta**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [guidelines](/okta/okta-oidc-js/blob/master/CONTRIBUTING.md#commit)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Adding Tests
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: 290951

## What is the new behavior?

Adds typescript declarations
Is an ES module

## Does this PR introduce a breaking change?
- [x] Yes
- [] No

Addition of typescript types may interfere with other code which has declared its own version of objects in this library. Those apps should remove their own definitions.

### CODE REVIEW

File: packages/configuration-validation/babel.config.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const presets = ['@babel/preset-env'];
+const plugins = [];
+
+// Process typescript when running in jest
+if (process.env.NODE_ENV === 'test') {
```
Comment (shuowu) [inline packages/configuration-validation/babel.config.js:5]: Does the `root` option work here? The test config can be moved into /test folder.

File: packages/configuration-validation/package.json [hunk 2]
```diff
@@ -32,17 +34,38 @@
...
+    "testEnvironment": "jsdom",
+    "testMatch": [
+      "**/test/*.{js,ts}"
+    ],
+    "roots": [
+      "test"
+    ]
   },
   "dependencies": {
+    "@okta/okta-auth-js": "^4.0.0",
```
Comment (shuowu) [inline packages/configuration-validation/package.json:68]: Looks like it only depends on types from `auth-js`, should it belongs to `dependencies` or `devDependenies`?

Comment (shuowu) [reply packages/configuration-validation/package.json:68]: never mind, looks like the module also exports new types based on the imported types.

File: packages/configuration-validation/babel.config.js (hunk 1 above)
Comment (aarongranick-okta) [reply packages/configuration-validation/babel.config.js:5]: The tests seem to work fine as .ts files using `babel-jest`. I'm not sure about moving the babel config. I copied this from okta-auth-js. I think the `root` option would apply if we were running the tests from the `/tests` directory (I think it  goes with **working** directory, not **src** directory

**Pull request merged**